cranelift/codegen/src/isa/x86/enc_tables.rs

```diff
@@ -139,6 +139,21 @@ fn size_plus_maybe_sib_or_offset_inreg1_plus_rex_prefix_for_inreg0_inreg1(
         + if needs_rex { 1 } else { 0 }
 }
 
+/// Calculates the size while inferring if the first and second input registers (inreg0, inreg1)
+/// require a dynamic REX prefix and if the second input register (inreg1) requires a SIB.
+fn size_plus_maybe_sib_inreg1_plus_rex_prefix_for_inreg0_inreg1(
+    sizing: &RecipeSizing,
+    enc: Encoding,
+    inst: Inst,
+    divert: &RegDiversions,
+    func: &Function,
+) -> u8 {
+    let needs_rex = (EncodingBits::from(enc.bits()).rex_w() != 0)
+        || test_input(0, inst, divert, func, is_extended_reg)
+        || test_input(1, inst, divert, func, is_extended_reg);
```

Comment:
Does this perhaps need to test some third register number to see if it has bit 3 set (== is_extended_reg, iiuc) ?  REX has 4 bits and it seems like you're only checking 3 here.

Reply:
The way I understand this is that each of these functions are specialized to the operands of the recipe (or instruction class, if you prefer). This function, for example, is only supposed to be used on recipes with `.operands_in(vec![fpr, gpr])` in which that `gpr` could be used with a SIB byte.

Reply:
So we wouldn't need to check for the third operand (REX.X?) in this case.

Reply:
Well, so long as you're sure that no third register can participate.

Reply:
😁 I'm only as sure as the filetests confirm. But, yes, this is how I understand the recipes/size calculator functions to work currently.

Reply:
The recipes that might call this encoding function ([here](https://github.com/bytecodealliance/wasmtime/blob/ac1e9d25241cf4a389431121ef86b29b336f65e0/cranelift/codegen/meta/src/isa/x86/recipes.rs#L1609)) takes an fpr and a gpr as inputs, so the second argument is always just a register. No memory operand with a third register here!